zulipterminal/helper.py

```diff
@@ -493,3 +493,141 @@ def notify(title: str, html_text: str) -> None:
     if command:
         res = subprocess.run(shlex.split(command), stdout=subprocess.DEVNULL,
                              stderr=subprocess.STDOUT)
+
+
+def parse_html_table(table_element: Any) -> Tuple[List[str], List[List[str]]]:
+    """
+    Parses an HTML table to extract the cell items and the column alignments.
+
+    The table cells are stored in `cells` in a row-wise manner.
+    cells = [[row0, row0, row0],
+             [row1, row1, row1],
+             [row2, row2, row2]]
+    """
+    headers = table_element.thead.tr.find_all('th')
+    rows = table_element.tbody.find_all('tr')
+    column_alignments = []
+
+    # Add +1 to count the header row as well.
+    cells = [[] for _ in range(len(rows) + 1)]  # type: List[List[str]]
+
+    # Fill up the `cells` with the header/0th row and extract alignments.
+    for header in headers:
+        cells[0].append(header.text)
+        column_alignments.append(header.get(('align'), 'left'))
+
+    # Fill up the `cells` with body rows.
+    for index, row in enumerate(rows, start=1):
+        for tdata in row.find_all('td'):
+            cells[index].append(tdata.text)
+    return (column_alignments, cells)
+
+
+TABLE_ROW = List[Union[str, Tuple[Optional[str], str]]]
+
+
+def pad_row_strip(row_strip: TABLE_ROW, fill_char: str=' ',
+                  fill_width: int=1) -> TABLE_ROW:
+    """
+    Returns back a padded row strip.
+
+    This only pads the box-drawing unicode characters. In particular, all the
+    connector characters are padded both sides, the leftmost character is
+    padded right and the rightmost is padded left.
+
+    The structure of `row_strip` for a table with two columns:
+    row_strip = [
+        leftmost_char,
+        cell_content,
+        connector_char,
+        cell_content,
+        connector_char,
+        rightmost_char,
+    ]
+    """
+    fill = fill_char * fill_width
+
+    # Pad the leftmost box-drawing character.
+    row_strip[0] = row_strip[0] + fill  # type: ignore
```

Comment:
I am not sure why `mypy` complains about these concatenations.

Reply:
I tried using `type: str` for `fill` but that didn't seem to solve the issue. I'll add comments regarding the ignores.